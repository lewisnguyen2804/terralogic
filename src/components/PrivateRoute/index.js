import React from 'react';
import { Route, Redirect } from 'react-router-dom';

const PrivateRoute = ({ component: Component, ...rest }) => (
    <Route {...rest} render={props => {
        const rememberUser = localStorage.getItem('rememberUser');
        let user = {};
        if (rememberUser === 'false') {
            user = sessionStorage.getItem('user');
        } else {
            user = localStorage.getItem('user');
        }

        return (rememberUser === 'true') ?
            <Component {...props} /> : (!user) ? 
            <Redirect to={{ pathname: '/login', state: { from: props.location } }} /> : <Component {...props} />;

    }} />
)

export default PrivateRoute;