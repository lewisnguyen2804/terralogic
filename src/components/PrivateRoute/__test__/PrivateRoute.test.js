import React from 'react'
import { shallow } from 'enzyme'
import PrivateRoute from '..';

describe('Test PrivateRoute', () => {
    it('Test PrivateRoute', () => {
        const privateRoute = shallow(<PrivateRoute />);
        expect(privateRoute).toMatchSnapshot()
    });
});