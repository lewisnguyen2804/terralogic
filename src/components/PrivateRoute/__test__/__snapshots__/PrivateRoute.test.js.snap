// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PrivateRoute Test what? 1`] = `ShallowWrapper {}`;
